test/HealthChecks.UI.Tests/Functional/UIApiRequestLimitingTests.cs

```diff
@@ -47,12 +47,12 @@ public async Task should_return_too_many_requests_status_code_when_exceding_conf
 
             using var server = new TestServer(webHostBuilder);
 
-            var requests = Enumerable.Range(1, maxActiveRequests)
-                .Select(n => server.CreateRequest($"/healthchecks-api").GetAsync());
+            var requests = Enumerable.Range(1, maxActiveRequests + 2)
+                .Select(_ => server.CreateRequest("/healthchecks-api").GetAsync()).ToList();
 
             var results = await Task.WhenAll(requests).ConfigureAwait(false);
 
-            results.Where(r => r.StatusCode == HttpStatusCode.TooManyRequests).Count().ShouldBe(requests.Count() - maxActiveRequests);

```

Comment:
Before fix `Count()` call executed `requests` enumerable. The same for `Where` leading to unpredicatable results.